packages/core/src/RPA/core/webdriver.py

```diff
@@ -5,26 +5,63 @@
 import platform
 import stat
 from pathlib import Path
+from types import MethodType
 from typing import List, Optional
 
 import requests
+from packaging import version
 from requests import Response
 from selenium import webdriver
 from selenium.webdriver.common.service import Service
 from selenium.webdriver.remote.webdriver import WebDriver
-from webdriver_manager.chrome import ChromeDriverManager
+from webdriver_manager.chrome import ChromeDriverManager as _ChromeDriverManager
 from webdriver_manager.core.download_manager import WDMDownloadManager
+from webdriver_manager.core.driver_cache import DriverCacheManager
 from webdriver_manager.core.http import WDMHttpClient
+from webdriver_manager.core.logger import log
 from webdriver_manager.core.manager import DriverManager
-from webdriver_manager.core.utils import ChromeType, get_browser_version_from_os
-from webdriver_manager.core.utils import os_name as get_os_name
+from webdriver_manager.core.os_manager import ChromeType, OperationSystemManager
+from webdriver_manager.drivers.chrome import ChromeDriver as _ChromeDriver
 from webdriver_manager.firefox import GeckoDriverManager
 from webdriver_manager.microsoft import EdgeChromiumDriverManager, IEDriverManager
 from webdriver_manager.opera import OperaDriverManager
 
 from RPA.core.robocorp import robocorp_home
 
 
+class ChromeDriver(_ChromeDriver):
+    """Custom class which correctly obtains the chromedriver download URL."""
+
+    def get_latest_release_version(self):
+        determined_browser_version = self.get_browser_version_from_os()
+        log(f"Get LATEST {self._name} version for {self._browser_type}")
+        if determined_browser_version:
+            parts = version.parse(determined_browser_version).release
+            if len(parts) == 4:
+                return determined_browser_version  # got a fully downloadable version
+
+        latest_release_url = (
+            self._latest_release_url
+            if (self._driver_version == "latest" or determined_browser_version is None)
+            else f"{self._latest_release_url}_{determined_browser_version}"
+        )
+        resp = self._http_client.get(url=latest_release_url)
+        return resp.text.rstrip()
+
+
+# FIXME(cmin764; 24 Jul 2023): Remove this when the following upstream Issue is solved:
+#  https://github.com/SergeyPirogov/webdriver_manager/issues/550
+class ChromeDriverManager(_ChromeDriverManager):
+    """Custom Chrome webdriver manager which correctly downloads the chromedriver."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.driver.get_latest_release_version = MethodType(
+            ChromeDriver.get_latest_release_version, self.driver
+        )
```

Comment:
This transforms an unbound method (attached to class) into a bound method (attached to instance), so we can transplant the new logic without dealing with a new instantiation, which would require the handling of all the input `kwargs` (and their defaults).